pkg/mover/chart.go

```diff
@@ -256,10 +262,12 @@ func (cm *ChartMover) loadChartFromIntermediateBundle(bundlePath string) error {
 // loadChartFromPath load the chart in memory from a given path
 func (cm *ChartMover) loadChartFromPath(path string) error {
 	var err error
-	if cm.chart, err = loader.Load(path); err != nil {
+	var chart *chart.Chart
+	if chart, err = loader.Load(path); err != nil {
 		return &ChartLoadingError{Path: path, Inner: err}
 	}
-	return nil
+	cm.chart, err = deduplicateChartFiles(chart)
```

Comment:
I'd return error before setting the value if it's set.

I mean

```
	if deduplicatedchart, err = deduplicateChartFiles(chart); err != nil {
         return err
        } else {
  cm.chart = deduplicatedchart
      }
```

otherwise you are setting the value even if it erroed

Reply:
Well, **we do want to set the chart even on error**.

This call has 2 possible exits but 3 different usages:
- Exit 1 no duplicates / Usage 1 use the chart.
- Exit 2 duplicates found error and fixed chart:
  - Usage 2 ignore display the error as a warning and use the chart.
  - Usage 3 error out and do not use the fixed chart.

What you propose would only make sense in usages 1&3. But for that case, if we are not going to use the fixed chart, we can avoid fixing the chart. This implementation keeps our options open and, to avoid having to add a flag, I default to use the fixed chart by default, not to block end users from using our charts.

Reply:
I see what you did there but again, I think this is overly complex for a requirement that does not exist. Why do we need to tell the user which is consuming the chart and probably is not the author of the chart that there are duplicated files inside? What action do you expect she to do with that warning?

having a chart with duplicated files in a tar is probably not that common (once we fix our pipeline) to deserve this amount of code in the core of this tool.